### PR TITLE
problems: fix anonymous users raising exceptions

### PR DESCRIPTION
**Pull request by cafehaine**
Should fix the crash occuring when visiting /train/submission-corrected/42 when not logged in

Cannot test ATM

### CODE REVIEW

File: prologin/problems/views.py [hunk 1]
```diff
@@ -411,6 +411,8 @@ class AjaxSubmissionCorrected(PermissionRequiredMixin, BaseDetailView):
     pk_url_kwarg = 'submission'
 
     def get_queryset(self):
+        if self.request.user.is_authenticated:
```
Comment (delroth) [inline prologin/problems/views.py]: Should be the opposite?

Comment (cafehaine) [reply prologin/problems/views.py]: ..duh, thanks

**Comment by seirl** [comment]
Shouldn't that just be a permission in the view instead of shoving that in get_queryset?

**Comment by cafehaine** [comment]
It should already have permissions in place, I don't really understand why get_queryset is being executed

**Pull request merged**